### PR TITLE
Automatically generate a service log when verify-egress detects blocked URLs

### PR DESCRIPTION
**Pull request by AlexVulaj**
Customers can often have a long list of blocked egress URLs. When SRE detects this, we then have to populate a servicelog template with each of the URLs individually. This is a tedious task that can require a ton of copy-paste.

This enhancement takes the output we already have, parses the URLs, and generates the servicelog to send without any human interaction needed.

**_As always, SLs generated by osdctl commands will still prompt for confirmation before sending the SL._**

[OSD-15300](https://issues.redhat.com/browse/OSD-15300)

### CODE REVIEW

File: cmd/network/verification.go [hunk 1]
```diff
@@ -129,12 +132,33 @@ func (e *EgressVerification) Run(ctx context.Context) {
 	e.log.Info(ctx, "Preparing to check %+v subnet(s) with network verifier.", len(inputs))
 
 	for i := range inputs {
-
 		e.log.Info(ctx, "running network verifier for subnet  %+v, security group %+v", inputs[i].SubnetID, inputs[i].AWS.SecurityGroupId)
 		out := onv.ValidateEgress(c, *inputs[i])
 		out.Summary(e.Debug)
-		if out.IsSuccessful() {
```
Comment (AlexVulaj) [inline cmd/network/verification.go:136]: The call to "ValidateEgress" above will already print this out on a successful run - no need to duplicate.

**Comment by clcollins** [comment]
/hold

For some unit tests/refactor

**Comment by clcollins** [comment]
/hold cancel
/approve
/lgtm

Thanks for going through and pairing with me while writing the tests for this!

File: cmd/network/verification.go [hunk 2]
```diff
@@ -129,14 +131,40 @@ func (e *EgressVerification) Run(ctx context.Context) {
...
+func generateServiceLog(out egressOutput, clusterId string) servicelog.PostCmdOptions {
+	failures, _, _ := out.Parse()
+	var failedEgresses []string
+	for _, failure := range failures {
+		failedEgresses = append(failedEgresses, strings.Split(failure.Error(), "Unable to reach ")[1])
+	}
+
+	if len(failedEgresses) > 0 {
+		return servicelog.PostCmdOptions{
+			Template:       "https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/required_network_egresses_are_blocked.json",
```
[the remark this answers is not in the document]
Comment (AlexVulaj) [reply cmd/network/verification.go:162]: Good call, commit incoming.

**Review by mjlshen — APPROVED** [review]
/lgtm

**Pull request merged**